scenes/game_elements/props/filling_barrel/components/filling_barrel.gd

```diff
@@ -34,35 +40,37 @@ func _set_color(new_color: Color) -> void:
 	if not is_node_ready():
 		return
 	if color:
-		sprite_2d.modulate = color
+		animated_sprite_2d.modulate = color
 	else:
-		sprite_2d.modulate = Color.WHITE
+		animated_sprite_2d.modulate = Color.WHITE
 
 
-func _set_texture(new_texture: Texture2D) -> void:
-	texture = new_texture
+func _set_sprite_frames(new_sprite_frames: SpriteFrames) -> void:
+	sprite_frames = new_sprite_frames if new_sprite_frames else DEFAULT_SPRITE_FRAMES
 	if not is_node_ready():
 		return
-	if texture:
-		sprite_2d.texture = texture
-	else:
-		sprite_2d.texture = DEFAULT_TEXTURE
+	if animated_sprite_2d:
+		animated_sprite_2d.sprite_frames = sprite_frames
+	update_configuration_warnings()
 
 
 func _ready() -> void:
 	_set_color(color)
-	_set_texture(texture)
+	_set_sprite_frames(sprite_frames)
+	if _are_sprite_frames_valid():
```

Comment:
I don't think we should add these kinds of validations. It will be better for a learner to see the error in the editor, than to hide the error, so they can fix it.